README.md

```diff
@@ -1 +1,228 @@
-# JavaScript Base-N Id Generator
+# JavaScript Base-N Id Generator _(bnid)_
+
+[![Build Status](https://travis-ci.org/digitalbazaar/bnid.png?branch=master)](https://travis-ci.org/digitalbazaar/bnid)
+
+> A JavaScript library for Web browsers and Node.js apps to generate random
+> ids and encode and decode them using various base-N encodings.
+
+## Table of Contents
+
+- [Background](#background)
+- [Install](#install)
+- [Usage](#usage)
+- [API](#api)
+- [Contribute](#contribute)
+- [Commercial Support](#commercial-support)
+- [License](#license)
+
+## Background
+
+This library provides tools for Web and Node.js to generate random ids and and
+encode and decode them in various base-N encodings. It came about from a need
+for a more compact encoding of ~128 bit ids than [UUID Version 4][] provided.
+
+The interface follows the [TextEncoder][]/[TextDecoder][] interfaces and
+provides [IdEncoder](#idencoder) and [IdDecoder](#iddecoder) classes. Instances
+can be configured with reusable encodings and other parameters. The encoder
+operates on a `Uint8Array` of input bytes. A [IdGenerator](#idgenerator) class
+is provided that can be used to generate random ids with selectable bit
+lengths, but any id data can be used.
+
+The encoder and decoder support various encodings and options:
+
+- `base16`/`hex`, `base16upper`: The simple [Base16][].
+- `base58`/`base58btc`: The [Base58][] Bitcoin alphabet as supported by
+  [base58-universal][].
+- Optional [multibase][] type prefix.
+- Fixed bit length. This is useful to ensure the output id length is constant
+  even when the id starts with an arbitrary number of zeros.
+
+### Alternatives
+
+There are
+*[many](https://www.npmjs.com/search?q=random%20id%20generator)*
+*[other](https://www.npmjs.com/search?q=id%20generator%)*
+*[random](https://www.npmjs.com/search?q=random%20id%)*
+id generators out there. Please do your research to find what best fits your
+needs.
+
+If you just want a small id generator with a customizable alphabet, start here:
+
+- [Nano ID](https://www.npmjs.com/package/nanoid)
```

Comment:
strike this section.

Reply:
Ok, just trying to be helpful to readers.